tests/e2e/testcases/icm/deploy/suite.go

```diff
@@ -0,0 +1,441 @@
+package deploy
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/interchain"
+	"github.com/ava-labs/avalanche-cli/tests/e2e/commands"
+	"github.com/ava-labs/avalanche-cli/tests/e2e/utils"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+const (
+	ewoqKeyName = "ewoq"
+	subnetName  = "testSubnet"
+)
+
+var _ = ginkgo.Describe("[ICM] deploy", func() {
+	ginkgo.Context("with valid input", func() {
+		ginkgo.BeforeEach(func() {
+			commands.StartNetwork()
+		})
+
+		ginkgo.AfterEach(func() {
+			commands.CleanNetwork()
+			err := utils.DeleteConfigs(subnetName)
+			gomega.Expect(err).Should(gomega.BeNil())
+			utils.DeleteCustomBinary(subnetName)
+		})
+		ginkgo.It("should deploy ICM contracts into c-chain", func() {
+			globalFlags := utils.GlobalFlags{
+				"local":             true,
+				"skip-update-check": true,
+			}
```

Comment:
globalFlags can be defined on top, so it doesn't have to be defined in every testcases